src/nifti/VolumeAcquisitionStreamer.js

```diff
@@ -89,20 +88,20 @@ export default class VolumeAcquisitionStreamer {
     if (fetcherData.headerPromise) {
       return fetcherData.headerPromise;
     }
+    const cachedVolume = this.volumeCache.get(imageIdObject);
```

Comment:
Is `imageIdObject` guaranteed to be unique across all timepoints? The original implementation seems to indicate the same image ID could occur within more than one timepoint.

Reply:
Here the imageIdObject is unique since it have timepoint also with this, but the caching is done by using the "file path" as key. This will result in using the same volume for each time point.
Here instead of current implementation, we can do the caching with "file path + time point" combination as key. But then there will be multiple volume entries for each time point, so we may need to introduce a limit for caching to avoid memory overload scenarios.

So removed the caching related changes in retrieval and return the "headerOnlyPromise"(this was initialized at the first header request) for each header request.